src/source/reporting.rst

```diff
@@ -0,0 +1,285 @@
+
+.. _time-range:
+
+Each time range clause is defined as follows:
+
+.. code:: javascript
+
+    {
+        "period":   <string>
+        "from":     <string>
+        "to":       <string>
+    }
+
+There are several *predefined* period values at your disposal: ``today``, ``yesterday``, ``thisWeek``, ``lastWeek``, ``thisMonth``, ``lastMonth``, ``thisYear``, and ``lastYear``, with their obvious meanings.
+If you use a predefined period for a particular time range, the *from* and *to* fields of that time range will be ignored.
+
+To use a *custom* period, you need to provide the start (*from*) and end (*to*) date (**both inclusive**) in the following format: ``YYYY-MM-DD``.
+The end date cannot be before the start one. For example:
+
+.. code:: javascript
+
+    "timeRanges": [
+        {
+                "period":   "custom",
+                "from":     "2017-01-01",
+                "to":       "2017-10-22"
+        }
+    ]
+
+
+Example with multiple time ranges, mixed custom and predefined:
+
+.. code:: javascript
+
+    "timeRanges": [
+        {
+            "period":   "thisWeek"
+        },
+        {
+            "period":   "custom",
+            "from":     "2017-01-08",
+            "to":       "2017-01-15"
+        }
+    ]
+
+
+Time Aggregation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _time-aggregation:
+
+The ``aggregate`` parameter is used in combination with the ``am:date`` dimension, and is **mandatory** in such case. It specifies the granularity/resolution of the time according to which the metrics data will be grouped in buckets. Possible values are: ``daily``, ``weekly``, ``quarterly``, ``montly``, ``yearly``.
+
+Filters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _filters:
+
+To request a subset of your data, use filters. For example, you can filter on particular category, region, or ad ID. Filters can currently be used on any of the supported dimensions (except for ``am:date``). In the request query, they should be provided as an array of filter clauses which filters out the requested data on particular field values. Each filter clause is of the following form:
+
+.. code:: javascript
+
+        {
+                "field":   <string>
+                "operator":  <string>
+                "value":  [<long>, ...]
+        }
+
+Each filter in the list adds more restriction on the result, i.e., there is implicit *AND* operation between them.
+The only valid ``operator`` at the moment is the set operator ``in``, and requires an array of values to be passed in the ``value`` field of the filter. For example:
+
+.. code:: javascript
+
+    "filters": [
+        {
+            "field":"am:adID",
+            "operator":"in",
+            "value": [1111,2222,3333]
+        },
+        {
+            "field":"am:regionID",
+            "operator":"in",
+            "value": [1234, 5678]
+        }
+    ]
+
+We plan to extend support for filters on metrics, as well as more comparison operators, if there is demand for it.
```

Comment:
not sure whether this line is necessary, this is a manual.

Reply:
just wanted to encourage feedback....removing